Documentation/contributing.rst

```diff
@@ -236,9 +240,54 @@ Runtime Tests
 After the new version of Cilium is running, you should run the runtime tests:
 
 ::
-   
+
     $ sudo make runtime-tests
 
+Ginkgo runtime tests
+^^^^^^^^^^^^^^^^^^^^
+
+The tests under the ``test/`` directory are built and run using the Ginkgo
+framework. Over time, all runtime tests will migrate to this framework. If
+you're new to Ginkgo, consider reading through the
+`Ginkgo Quickstart <https://onsi.github.io/ginkgo/#getting-started-writing-your-first-test>`_
+guide to understand how to write tests.
+
+These test scripts will invoke ``vagrant`` to create virtual machine(s) to
+run the tests. The tests make heavy use of the Ginkgo ``focus`` concept to
+determine which VMs are necessary to run particular tests. All test names
+*must* begin with one of the following prefixes:
+
+* ``Runtime``: Test cilium in a runtime environment running on a single node.
+* ``K8s``: Create a small multi-node kubernetes environment for testing
+  features beyond a single host, and for testing kubernetes-specific features.
+
+Running all of the Ginkgo tests may take an hour or longer. To run all the
+ginkgo tests, invoke the make command as follows:
```

Comment:
"as follows from the root of the cilium repository:"

Reply:
Thanks, done.